staging/src/k8s.io/apiextensions-apiserver/pkg/controller/status/naming_controller.go

```diff
@@ -238,19 +238,24 @@ func (c *NamingConditionController) sync(key string) error {
 		return err
 	}
 
-	acceptedNames, namingCondition, establishedCondition := c.calculateNamesAndConditions(inCustomResourceDefinition)
+	// Skip checking names if Spec and Status names are same.
+	if reflect.DeepEqual(inCustomResourceDefinition.Spec.Names, inCustomResourceDefinition.Status.AcceptedNames) {
```

Comment:
use `semantic.DeepEqual` instead (it follow our semantics that `nil` and empty lists are the same logically).